mvpa2/measures/rsa.py

```diff
@@ -225,8 +233,19 @@ def _call(self,dataset):
         dsm = pdist(data,self.params.pairwise_metric)
         if self.params.comparison_metric=='spearman':
             dsm = rankdata(dsm)
-        rho, p = pearsonr(dsm,self.target_dsm)
-        if self.params.corrcoef_only:
-            return Dataset([rho], fa={'metrics': ['rho']})
-        else:
-            return Dataset([[rho,p]], fa={'metrics': ['rho', 'p']})
+        if self.control_dsms == None:
+            rho, p = pearsonr(dsm,self.target_dsm)
+            if self.params.corrcoef_only:
+                return Dataset([rho], fa={'metrics': ['rho']})
+            else:
+                return Dataset([[rho,p]], fa={'metrics': ['rho', 'p']})
+        elif self.control_dsms != None:
+            X = np.column_stack([np.array([1. for i in self.target_dsm])]+[self.target_dsm]+self.control_dsms)
+            res = OLS(endog=dsm,exog=X).fit()
+            rho = res.params[1]*(np.std(X[:,1])/np.std(dsm))
+            p = res.pvalues[1]
+            if self.params.corrcoef_only:
+                return Dataset([rho], fa={'metrics': ['rho']})
+            else: 
+                return Dataset([[rho,p]], fa={'metrics': ['rho','p']})
```

Comment:
This last block (`if self...`) is identical to if we have control_dsms or not, so move it out from the "if self.control_dsms is None`.  No need for elif here, else would suffice ;)